### PR TITLE
Fish 470 mdb nullpointer when customized mdb pool

### PR DESCRIPTION
**Pull request by breakponchito**
## Description

this is a fix to sanitize the NullPointerException when customizing properties for the MDB 
### New tests

I added unit test for the changes on the following classes:  BaseContainer and JCDIServiceImpl.  In the case of the unit test for the class MessageBeanContainer I omitted because of the complexity to deal with static methods
### Testing Performed

I already tested with unit testing and also on the local environment, please review the following comment to verify the result of tests: https://payara.atlassian.net/browse/FISH-470?focusedCommentId=57455
### Testing Environment

Windows 10 with JDK 8

### CODE REVIEW

**Comment by breakponchito** [comment]
Jenkins test

**Comment by breakponchito** [comment]
Jenkins test

**Comment by breakponchito** [comment]
Jenkins test

**Comment by breakponchito** [comment]
Jenkins test

File: appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java [hunk 1]
```diff
@@ -2058,7 +2063,11 @@ public void preInvoke(EjbInvocation inv) {
 
     public boolean intercept(CallbackType eventType, EJBContextImpl ctx)
             throws Throwable {
-
+        //sanitizing the null reference of interceptorManager
+        if (interceptorManager == null) {
+            _logger.severe("the reference for interceptorManager is not available, this is an un-sync state of the container");
```
Comment (Pandrex247) [inline appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java:2068]:
```suggestion
            _logger.severe("The reference for interceptorManager is not available, this is an un-sync state of the container");
```

File: appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/MessageBeanContainer.java [hunk 2]
```diff
@@ -799,7 +799,11 @@ private MessageBeanContextImpl createMessageDrivenEJB() throws CreateException {
 
             // Call ejbCreate OR @PostConstruct on the bean.
             intercept(POST_CONSTRUCT, context);
-
+            //sanitizing the null reference of ejbProbeNotifier and returning null
+            if(ejbProbeNotifier == null) {
+                _logger.severe("the reference for ejbProbeNotifier is not available, this is an un-sync state of the container");
```
Comment (Pandrex247) [inline appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/MessageBeanContainer.java:804]:
```suggestion
                _logger.severe("The reference for ejbProbeNotifier is not available, this is an un-sync state of the container");
```

File: appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/JCDIServiceImpl.java [hunk 3]
```diff
@@ -213,7 +213,12 @@ public void setELResolver(ServletContext servletContext) throws NamingException
 
         WeldBootstrap bootstrap = weldDeployer.getBootstrapForApp(ejb.getEjbBundleDescriptor().getApplication());
         WeldManager weldManager = bootstrap.getManager(bda);
-
+        //sanitizing the null reference of weldManager and returning null
+        //when calling _createJCDIInjectionContext
+        if(weldManager == null) {
+            logger.severe("the reference for weldManager is not available, this is an un-sync state of the container");
```
Comment (Pandrex247) [inline appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/JCDIServiceImpl.java:219]:
```suggestion
            logger.severe("The reference for weldManager is not available, this is an un-sync state of the container");
```

File: appserver/web/weld-integration/src/test/java/org/glassfish/weld/WeldActivatorTest.java [hunk 4]
```diff
@@ -56,6 +56,8 @@ public void testAll() throws Exception {
 
         Field instanceField = SingletonProvider.class.getDeclaredField("INSTANCE");
         instanceField.setAccessible(true);
+        //nullifyng the internal reference of the SingletonProvider to not broke the jenkins execution
```
Comment (Pandrex247) [inline appserver/web/weld-integration/src/test/java/org/glassfish/weld/WeldActivatorTest.java:59]:
```suggestion
        //nullifyng the internal reference of the SingletonProvider to not bleed across tests
```

**Review by Pandrex247 — COMMENTED** [review]
Slight pedantry, but log statements should start with a capital letter - they are sentences after all.

File: appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java (hunk 1 above)
Comment (breakponchito) [reply appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java]: already done

File: appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/MessageBeanContainer.java (hunk 2 above)
Comment (breakponchito) [reply appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/MessageBeanContainer.java]: already done

File: appserver/web/weld-integration/src/test/java/org/glassfish/weld/WeldActivatorTest.java (hunk 4 above)
Comment (breakponchito) [reply appserver/web/weld-integration/src/test/java/org/glassfish/weld/WeldActivatorTest.java]: already done

File: appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/JCDIServiceImpl.java (hunk 3 above)
Comment (breakponchito) [reply appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/JCDIServiceImpl.java]: already done

**Comment by breakponchito** [comment]
Jenkins test

**Review by Pandrex247 — APPROVED** [review]

File: appserver/ejb/ejb-container/src/test/java/com/sun/ejb/containers/BaseContainerTest.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package com.sun.ejb.containers;
+
+import com.sun.enterprise.deployment.LifecycleCallbackDescriptor;
+import junit.framework.TestCase;
```
Comment (pzygielo) [inline appserver/ejb/ejb-container/src/test/java/com/sun/ejb/containers/BaseContainerTest.java:43]: Why?

Comment (breakponchito) [reply appserver/ejb/ejb-container/src/test/java/com/sun/ejb/containers/BaseContainerTest.java:43]: We added this to test the changes we added to the BaseContainer component

Comment (pzygielo) [reply appserver/ejb/ejb-container/src/test/java/com/sun/ejb/containers/BaseContainerTest.java:43]: Why this JUnit3 import?

**Pull request merged**